trio/_tools/gen_exports.py

```diff
@@ -94,18 +108,41 @@ def create_passthrough_args(funcdef: ast.FunctionDef | ast.AsyncFunctionDef) ->
     return "({})".format(", ".join(call_args))
 
 
-def gen_public_wrappers_source(source_path: Path, lookup_path: str) -> str:
+def gen_public_wrappers_source(source_path: Path | str, lookup_path: str) -> str:
     """Scan the given .py file for @_public decorators, and generate wrapper
     functions.
 
     """
     generated = [HEADER]
+
+    # This is only triggered by check.sh, not test_gen_exports.py
+    if lookup_path == "runner.io_manager" and isinstance(
+        source_path, Path
+    ):  # pragma: no coverage
+        for file_indicator, platform in (
+            ("windows", "win32"),
+            ("kqueue", "darwin"),
+            ("epoll", "linux"),
```

Comment:
Instead of doing a redundant lookup here, it'd be better to change the `sources_and_lookups`/`to_wrap` to also have the platform name. Maybe also change the type to a attrs class, so we can use keywords to identify the different strings passed in.